crates/rnote-ui/src/dialogs/mod.rs

```diff
@@ -552,6 +553,38 @@ pub(crate) async fn dialog_edit_selected_workspace(appwindow: &RnAppWindow) {
     dialog.present(appwindow);
 }
 
+pub(crate) async fn dialog_trash_file(appwindow: &RnAppWindow, filerow: &RnFileRow) -> () {
+    let builder = Builder::from_resource(
+        (String::from(config::APP_IDPATH) + "ui/dialogs/dialogs.ui").as_str(),
+    );
+    let dialog: adw::AlertDialog = builder.object("dialog_trash_file").unwrap();
+
+    match dialog.choose_future(appwindow).await.as_str() {
+        "trash" => {
+            glib::spawn_future_local(clone!(@weak filerow, @weak appwindow => async move {
+                let Some(current_file) = filerow.current_file() else {
+                    return;
+                };
+                current_file.trash_async(
+                    glib::source::Priority::DEFAULT,
+                    None::<&gio::Cancellable>,
+                    clone!(@weak filerow, @strong current_file => move |res| {
+                        if let Err(e) = res {
+                            appwindow.overlays().dispatch_toast_error(&gettext("Trashing file failed"));
+                            debug!("Trash filerow file `{current_file:?}` failed , Err: {e:?}");
```

Comment:
This should be an error event

Reply:
I was thinking the same, that's however how the current trash action handled it, which is why I kept this behaviour. I'll change that.

Reply:
thanks!